chapter_11.md

```diff
@@ -1,16 +1,202 @@
+    - `.tmp`, this will hold the actual snapshots and when there are diffs they will be stored here
+    - `__snapshots__`, this will hold the baseline images. If there is no baseline image it will automatically be added. It will hold an `example`-folder (the name of the Storybook project) and in that folder you will find the components, each holder a folder of the browser you have been using. By default this is Chrome in headless mode
+6. The images have automatically been clipped to the "estimated" size of the component. If you want you can also:
+    - select the [browsers](https://github.com/webdriverio/visual-testing/tree/main?tab=readme-ov-file#--browsers) by adding `--browsers=chrome,firefox`
+    - select an [emulated device mode](https://github.com/webdriverio/visual-testing/tree/main?tab=readme-ov-file#--devices) by for example providing for example `--devices="iPhone 14 Pro Max","Pixel 3 XL"` to the command.
+
+> [!TIP]
+> If you want to see what is happening then provide `--headless=false` to the command line so the test will run in "normal" mode
+
+#### External hosted Storybook instance
+
+For this test we are going to use an externally hosted Storybook instance at [https://govuk-react.github.io/govuk-react/](https://govuk-react.github.io/govuk-react/). To make it a bit faster we will use sharding on 10 instances
+
+1. Run the following command `npx wdio ./wdio.conf.ts --storybook --url=https://govuk-react.github.io/govuk-react/ --numShards=10`
+2. The outcome should be that you will have around 200 snapshots taken in less than 20 seconds. In that time each screenshot would be stored as a new baseline. You can check the `__snapshots__`-folder at the root of the project to validate them.
+3. Now run the command `npx wdio ./wdio.conf.ts --storybook --url=https://govuk-react.github.io/govuk-react/ --numShards=10` again to "really" check all snapshots.
+4. The outcome would be that a few snapshots fail, see the `./.tmp/diff/desktop_chrome-headless-shell` folder. 99.99% sure this is related to a spinner.
+5. There are cases that you want to exclude certain components, this can be done by providing the option `--skipStories={string}` through the command line. This can be the component name (which you can find in the URL of the hosted Storybook instance, for example `https://govuk-react.github.io/govuk-react/?path=/story/utility-icons-spinner--spinner` where `utility-icons-spinner--spinner` is the name of the component), or a regular expression that would match multiple components. Now run the following command where we exclude the spinner components through a regular expression `npx wdio ./wdio.conf.ts --storybook --url=https://govuk-react.github.io/govuk-react/ --numShards=10 --skipStories="/.*(loading-box|spinner).*/"`
+6. Now 6 components have been skipped and all tests should succeed
+
+## Mobile Native App
```

Comment:
I assume there will be more text here.

Reply:
Yes, got the text ready, will push it tomorrow